db/migrate/20150416095248_create_tag_associations.rb

```diff
@@ -0,0 +1,10 @@
+class CreateTagAssociations < ActiveRecord::Migration
+  def change
+    create_table :tag_associations do |t|
+      t.integer :from_tag_id, index: true, null: false
+      t.integer :to_tag_id, index: true, null: false
+
+      t.timestamps
+    end
+  end
```

Comment:
We could do with some indexes here. A unique index on `[:from_tag_id, :to_tag_id]` would be good to ensure data integrity.  We'd also then want one on `to_tag_id` to allow efficiently traversing in the reverse direction.